src/pkg/build/conveyorPacker_busybox.go

```diff
@@ -0,0 +1,158 @@
+
+	c.recipe = recipe
+
+	c.b, err = NewBundle("")
+	if err != nil {
+		return
+	}
+
+	//get mirrorURL, OSVerison, and Includes components to definition
+	mirrorurl, ok := recipe.Header["mirrorurl"]
+	if !ok {
+		return fmt.Errorf("Invalid busybox header, no MirrurURL specified")
+	}
+
+	err = c.insertBaseEnv()
+	if err != nil {
+		return fmt.Errorf("While inserting base environment: %v", err)
+	}
+
+	err = c.insertBaseFiles()
+	if err != nil {
+		return fmt.Errorf("While inserting files: %v", err)
+	}
+
+	busyBoxPath, err := c.insertBusyBox(mirrorurl)
+	if err != nil {
+		return fmt.Errorf("While inserting busybox: %v", err)
+	}
+
+	cmd := exec.Command(busyBoxPath, `--install`, filepath.Join(c.b.Rootfs(), "/bin"))
+
+	sylog.Debugf("\n\tBusyBox Path: %s\n\tMirrorURL: %s\n", busyBoxPath, mirrorurl)
+
+	err = cmd.Run()
+	if err != nil {
+		return fmt.Errorf("While performing busybox install: %v", err)
+	}
+
+	return nil
+}
+
+// Pack puts relevant objects in a Bundle!
+func (cp *BusyBoxConveyorPacker) Pack() (b *Bundle, err error) {
+
+	err = cp.insertRunScript()
+	if err != nil {
+		return nil, fmt.Errorf("While inserting base environment: %v", err)
+	}
+
+	cp.b.Recipe = cp.recipe
+
+	return cp.b, nil
+}
+
+func (c *BusyBoxConveyor) insertBaseFiles() (err error) {
+
+	ioutil.WriteFile(filepath.Join(c.b.Rootfs(), "/etc/passwd"), []byte("root:!:0:0:root:/root:/bin/sh"), 0664)
+	if err != nil {
+		return
+	}
+
+	ioutil.WriteFile(filepath.Join(c.b.Rootfs(), "/etc/group"), []byte(" root:x:0:"), 0664)
+	if err != nil {
+		return
+	}
+
+	ioutil.WriteFile(filepath.Join(c.b.Rootfs(), "/etc/hosts"), []byte("127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4"), 0664)
+	if err != nil {
+		return
+	}
+
+	return
+}
+
+func (c *BusyBoxConveyor) insertBusyBox(mirrorurl string) (busyBoxPath string, err error) {
+
+	os.Mkdir(filepath.Join(c.b.Rootfs(), "/bin"), 0755)
+
+	resp, err := http.Get(mirrorurl)
+	if err != nil {
+		return "", fmt.Errorf("While performing http request: %v", err)
+	}
+	defer resp.Body.Close()
+
+	f, err := os.Create(filepath.Join(c.b.Rootfs(), "/bin/busybox"))
+	if err != nil {
+		return
+	}
+	defer f.Close()
+
+	bytesWritten, err := io.Copy(f, resp.Body)
+	if err != nil {
+		return
+	}
+
+	//Simple check to make sure file received is the correct size
+	if bytesWritten != resp.ContentLength {
+		return "", fmt.Errorf("File received is not the right size. Supposed to be: %v  Actually: %v", resp.ContentLength, bytesWritten)
+	}
+
+	err = os.Chmod(f.Name(), 0755)
+	if err != nil {
+		return
+	}
+
+	return filepath.Join(c.b.Rootfs(), "/bin/busybox"), nil
+}
+
+func (c *BusyBoxConveyor) insertBaseEnv() (err error) {
+	if err = makeBaseEnv(c.b.Rootfs()); err != nil {
+		return
+	}
+	return nil
+}
+
+func (cp *BusyBoxConveyorPacker) insertRunScript() (err error) {
+
+	ioutil.WriteFile(filepath.Join(cp.b.Rootfs(), "/.singularity.d/runscript"), []byte("#!/bin/sh\n"), 0755)
```

Comment:
I just checked against 2.x... - that gives a runscript like this from the example def file for busybox.

```
#!/bin/sh 

    echo "Running command: $*"
    exec "$@"
```

Reply:
So I wasnt planning on these conveyorPackers to have 2x functionality yet because Definition handling hasnt been completely sorted out. My plan was to get building valid fs's working first with runscripts that just open a shell for now so that i know we can build from all these sources and then make correct runscripts when I implement proper Definition file handling